packages/camera/lib/new/src/camera_controller.dart

```diff
@@ -88,6 +92,22 @@ class CameraController {
   ///
   /// This will choose the best [CameraAPI] for the current device.
   static Future<List<CameraDescription>> availableCameras() async {
+    final List<CameraDescription> devices = <CameraDescription>[];
+
+    final DeviceInfoPlugin infoPlugin = DeviceInfoPlugin();
+    if (defaultTargetPlatform == TargetPlatform.android) {
```

Comment:
subjective, optional: Consider using guard clauses and early returns to reduce nesting here.

https://testing.googleblog.com/2017/06/code-health-reduce-nesting-reduce.html

```dart
if (defaultTargetPlatform != TargetPlatform.android || await infoPlugin.androidInfo >= 21) {
   throw Unimplemented Error;
}

final int numCameras = await Camera.getNumberOfCameras();
// etc
```

Reply:
I wrote it like this since it will eventually expand to:

```dart
if (defaultTargetPlatform == TargetPlatform.android) {
  final AndroidDeviceInfo info = await infoPlugin.androidInfo;
  if (info.version.sdkInt < 21) {
    // Camera Implementation
  } else {
    // Camera2 Implementation
  }
} else if (defaultTargetPlatform == TargetPlatform.iOS) {
  // ios implementation
}
```

Is there another way to prevent the nesting without making another method?

Reply:
Ah okay. Yes, actually! You can flatten nested ifs like that by calculating the final logical cases first, and then casing into each of them. [Explanation](https://softwareengineering.stackexchange.com/a/205814).

```dart
final bool isAndroid = defaultTargetPlatform == TargetPlatform.android;
final bool isAndroidSdkBelow21 = isAndroid && (await infoPlugin.androidInfo).version.sdkInt < 21;
final bool isIos = !isAndroid && defaultTargetPlatform == TargetPlatform.ios;

if (isAndroidSdkBelow21) {
  // Camera
} else if (isAndroid && !isAndroidSdkBelow21) {
  // Camera 2
} else if (isIos) {
  // ios
} else {
  // throw
}
```

I think smaller methods here would also be alright. I am personally very averse to nesting. This is just a subjective readability suggestion, feel free to ignore it.